packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -284,6 +284,31 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     `)
   }  
 
+  async getUsersWithNewDialogueChecks(minutes: number): Promise<DbUser[]> {
+    return this.manyOrNone(`
+      SELECT "Users".*
+      FROM "Users"
+      INNER JOIN "DialogueChecks" ON "Users"._id = "DialogueChecks"."targetUserId"
+      WHERE
+          "DialogueChecks"."checkedAt" > NOW() - INTERVAL '$1 minutes'
+          AND "DialogueChecks".checked = true
+          AND NOT EXISTS (
```

Comment:
This clause should probably be a self-join instead, but not blocking

Reply:
what's the gain of a self join? I guess if you're looking at a table more than once it can be handy?